tokenserver-db/src/lib.rs

```diff
@@ -1,10 +1,14 @@
 extern crate diesel;
-#[macro_use]
 extern crate diesel_migrations;
 #[macro_use]
 extern crate slog_scope;
 
-mod error;
+use diesel::r2d2::{ConnectionManager, PooledConnection};
```

Comment:
I wonder if we should add a feature check here, kind of like we do for spanner. One of the features should be selected (either "mysql" or "sqlite") because otherwise things are not going to work well at all.

Reply:
Do you know of a way to force the user to choose a feature ? Something like a compile time panic if no feature is enabled.

Reply:
build.rs
```rust
If ![cfg!(feature = "feat1"), cfg!(feature = "feat2")].iter().any(){
    panic("You need to enable at least one feature")
}
```

Im on my phone atm, so idk if this compiles, but it seems reasonable

Reply:
Ah! Sorry, I thought you saw the trick we do in [syncstorage-db](https://github.com/mozilla-services/syncstorage-rs/blob/master/syncstorage-db/src/lib.rs#L37C1-L37C57) 

```
#[cfg(not(any(feature = "mysql", feature = "spanner")))]
```
basically, just add `sqlite` into that mix. There's also a line above that which does a check to make sure you don't specify more than one storage engine.

Reply:
Turns out I already added this to [syncstorage-db](https://github.com/eragonfr/syncstorage-rs/blob/syncstorage-sqlite/syncstorage-db/src/lib.rs#L46-L47)
```rust
#[cfg(not(any(feature = "mysql", feature = "spanner", feature = "sqlite")))]
compile_error!("exactly one of the \"mysql\", \"spanner\" and \"sqlite\" features must be enabled");
```

Do you want the same thing for tokenserver-db ?

Reply:
It's not a bad idea.

The long(-ish) term idea is that we eventually roll tokenserver into syncstorage so that there's one less weird thing hanging out on it's own. Tokenserver will definitely stay MySQL based for the forseeable future, but I don't know if you want to do the same work to convert it to Sqlite. (Much of tokenserver deals with juggling users between various "nodes" and "clusters" which can be abandoned, even for many stand-alone systems. You could also leave that as an exercise for some future person, and just make notes about the things that would need changing.)

Reply:
I added the feature checks in [d8c556b](https://github.com/mozilla-services/syncstorage-rs/pull/1520/commits/d8c556bb49e0c379d9e05b8637ba5af1cf8f3e5c)

I already almost have tokenserver fully working on SQLite. There's only 8 tests that still fails. But it should be pretty easy to fix.